lib/index.js

```diff
@@ -59,6 +59,7 @@ function AssetGraph(options) {
 
     // this.root might be undefined, in which case urlTools.urlOrFsPathToUrl will use process.cwd()
     this.root = normalizeUrl(urlTools.urlOrFsPathToUrl(this.root, true)); // ensureTrailingSlash
+    this.canonicalRoot = normalizeUrl(urlTools.urlOrFsPathToUrl(this.canonicalRoot, true)); // ensureTrailingSlash
```

Comment:
Those are the only examples I could come up with as well. Possibly with the `//` prefix on the domain.

I'm still a bit unsure if supporting all of these syntaxes is preferable over just a simple substring match. If we add the source expression match I think it's important that we can make fully reversible operations if a relation switches from canonical to local, back to canonical. In that situation I would prefer the href to end up exactly as originally written